sway/commands/input.c

```diff
@@ -0,0 +1,65 @@
+#include <string.h>
+#include <strings.h>
+#include "sway/commands.h"
+#include "sway/input/input-manager.h"
+#include "log.h"
+
+struct cmd_results *cmd_input(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "input", EXPECTED_AT_LEAST, 2))) {
+		return error;
+	}
+
+	if (config->reading && strcmp("{", argv[1]) == 0) {
```

Comment:
The `if (argc > 2)` check should be done before this line and return an error.

Reply:
Oh, there's already `checkarg(argc, "input", EXPECTED_AT_LEAST, 2))`. So just remove the check, I guess?

Reply:
I think the checkarg needs to expect 3 arguments for `seat <name> <command>` or `seat <name> {`, then the argc check will be useless. Maybe this was copied over from the back block, which doesn't require an identifier.

Reply:
I don't think the command name is included in `argc`, is it?

Reply:
argc is 1 for a command with no argument with just argv[0] set, so yes it is. You'll have 1 + number of arguments.

Reply:
I'm not sure about it. Right below, `argv[0]` is used as the device identifier and `argv[1]` is `{`.

Reply:
Nevermind, it gets the original args minus the first command. Going to fix it by doing another checkarg so a command like `seat <name> attach` is not valid.

Reply:
Yeah this is confusing. In sway's command handlers argv[0] is the first argument and argv[-1] is the command name.